pkg/xray/reports.go

```diff
@@ -0,0 +1,802 @@
+	m := filter.List()[0].(map[string]interface{})
+
+	filters.Unknown = m["unknown"].(bool)
+	filters.Unrecognized = m["unrecognized"].(bool)
+
+	filters.LicenseNames = util.CastToStringArr(m["license_names"].(*schema.Set).List())
+	filters.LicensePatterns = util.CastToStringArr(m["license_patterns"].(*schema.Set).List())
+
+	return &filters
+}
+
+func unpackOperationalRisksFilters(filter *schema.Set) *Filters {
+	var filters Filters
+	m := filter.List()[0].(map[string]interface{})
+
+	if m["component"] != nil {
+		filters.Component = m["component"].(string)
+	}
+	if m["artifact"] != nil {
+		filters.Artifact = m["artifact"].(string)
+	}
+
+	filters.Risks = util.CastToStringArr(m["risks"].(*schema.Set).List())
+
+	if m["scan_date"] != nil {
+		filters.ScanDate = unpackStartAndEndDate(m["scan_date"].(*schema.Set))
+	}
+
+	return &filters
+}
+
+func unpackCvssScore(d *schema.Set) *CvssScore {
+	var cvssScore CvssScore
+
+	if len(d.List()) > 0 {
+		f := d.List()[0].(map[string]interface{})
+		cvssScore = CvssScore{
+			MinScore: f["min_score"].(float64),
+			MaxScore: f["max_score"].(float64),
+		}
+		return &cvssScore
+	}
+
+	return nil
+}
+
+func unpackStartAndEndDate(d *schema.Set) *StartAndEndDate {
+	var dates StartAndEndDate
+
+	if len(d.List()) > 0 {
+		f := d.List()[0].(map[string]interface{})
+		dates = StartAndEndDate{
+			Start: f["start"].(string),
+			End:   f["end"].(string),
+		}
+		return &dates
+	}
+
+	return nil
+}
+
+func resourceXrayVulnerabilitiesReportCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+	return createReport("vulnerabilities", d, m)
+}
+
+func resourceXrayLicensesReportCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+	return createReport("licenses", d, m)
+}
+
+func resourceXrayViolationsReportCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+	return createReport("violations", d, m)
+}
+
+func resourceXrayOperationalRisksReportCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+	return createReport("operationalRisks", d, m)
+}
+
+func resourceXrayReportRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	report := Report{}
+
+	projectKey := d.Get("project_key").(string)
+	req, err := getRestyRequest(m.(*resty.Client), projectKey)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	resp, err := req.
+		SetResult(&report).
+		SetPathParams(map[string]string{
+			"reportId": d.Id(),
+		}).
+		Get("xray/api/v1/reports/{reportId}")
+	if err != nil {
+		if resp != nil && resp.StatusCode() == http.StatusNotFound {
+			tflog.Warn(ctx, fmt.Sprintf("Xray report (%s) not found, removing from state", d.Id()))
+			d.SetId("")
+		}
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func resourceXrayReportDelete(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	report := unpackReportProjectKey(d)
```

Comment:
Why is this func necessary? In line #629 it's just a single line, whereas here we have an extra func that only used here, one time.

Reply:
Yes, you are right, removed.